app/lib/actions/remote_execution/run_hosts_job.rb

```diff
@@ -75,7 +75,12 @@ def finalize
         job_invocation.save!
 
         Rails.logger.debug "cleaning cache for keys that begin with 'job_invocation_#{job_invocation.id}'"
-        Rails.cache.delete_matched(/\A#{JobInvocation::CACHE_PREFIX}_#{job_invocation.id}/)
+        delete_query = /\A#{JobInvocation::CACHE_PREFIX}_#{job_invocation.id}/
+
+        if redis_cache?
+          delete_query = "#{JobInvocation::CACHE_PREFIX}_#{job_invocation.id}*"
+        end
+        Rails.cache.delete_matched(delete_query)
```

Comment:
Could we do it a little bit differently? Rather than building the query inside `#finalize`, we'd have a method (say `cache_deletion_query`) which would return the right query for the currently configured cache store and in here we'd just do something like `Rails.cache.delete_matched(cache_deletion_query(job_invocation.id))`?